model/secret.py

```diff
@@ -57,7 +57,7 @@ def key_from_gen(self) -> bytes:
     def cipher_algorithm(self):
         return Cipher(self.raw.get('cipher_algorithm'))
 
-    def generation(self):
+    def generation(self) -> int:
         return self.raw.get('generation')
```

Comment:
nit: while you are at it, you might add `int(...)` around the return to validate it really _is_ an int